packages/ddp-server/livedata_server.js

```diff
@@ -191,11 +193,49 @@ Object.assign(Session.prototype, {
     return this._isSending || !this.server.getPublicationStrategy(collectionName).useCollectionView;
   },
 
+  _batchedAdds: new Map(),
 
   sendAdded(collectionName, id, fields) {
-    if (this._canSend(collectionName)) {
-      this.send({ msg: 'added', collection: collectionName, id, fields });
+    if (!this._canSend(collectionName)) {
+      return;
+    }
+
+    if (!this._batchedAdds.has(collectionName)) {
+      this._batchedAdds.set(collectionName, {
+        docs: [],
+        timeoutHandle: null
+      });
     }
+
+    const batch = this._batchedAdds.get(collectionName);
+
+    batch.docs.push({ id, fields });
+
+    if (batch.timeoutHandle) {
+      clearTimeout(batch.timeoutHandle);
+    }
+
+    const sendBatch = () => {
+      try {
+        this.sendAddedBatch(collectionName, batch.docs);
+        this._batchedAdds.delete(collectionName);
+      } catch (e) {
+        console.error('error sending batch', e)
+      }
+    };
+
+    if (batch.docs.length >= 20) {
+      sendBatch();
+    } else {
+      batch.timeoutHandle = setTimeout(() => {
+        console.log('send batch timeout', this._batchedAdds.size)
+        sendBatch();
+      }, 5000);
+    }
+  },
+
+  sendAddedBatch(collectionName, docs) {
+    this.send({ msg: 'addedBatch', collection: collectionName, docs });
   },
 
   sendChanged(collectionName, id, fields) {
```

Comment:
Maybe it needs to change `sendChanged` to be batched too?

Reply:
The testing scenario produced no changed messages, and seems to be tied to the message sized, we tested many scenarios, with/without async, and only occurred when the batch was a certain size 😞